src/py42/clients/alerts.py

```diff
@@ -136,3 +136,30 @@ def get_aggregate_data(self, alert_id):
             :class:`py42.response.Py42Response`
         """
         return self._alert_service.get_aggregate_data(alert_id)
+
+    def get_all_alert_details(self, query):
+        """
+        Helper method that combines :func:`.search_all_pages()` and :func:`.get_details()`.
+        Returns an iterator of alert detail objects in chronological order by alert
+        creation date.
+
+        Args:
+            query (:class:`py42.sdk.queries.alerts.alert_query.AlertQuery`): An alert query.
+            ascending (bool): Determines if sort ordering should be ascending/descending
+                based on alert creation time (overrides the sort order on the query object).
+                Defaults to True.
+
+        Returns:
+            generator: An object that iterates over alert detail items.
+        """
+        query.page_size = 100
+        sort_key = query.sort_key[0].lower() + query.sort_key[1:]
+        reverse = query.sort_direction == "desc"
+        pages = self._alert_service.search_all_pages(query)
+        for page in pages:
+            alert_ids = [alert["id"] for alert in page["alerts"]]
+            alert_details = self._alert_service.get_details(alert_ids)
+            for alert in sorted(
+                alert_details["alerts"], key=lambda x: x[sort_key], reverse=reverse,
```

Comment:
does this behave as expected when the field referenced by sort_key is a timestamp? I feel like python would just treat it as a string unless we explicitly convert.

Reply:
oh I see, we're using a sane format where those happen to end up with the same result, cool.